src/storage/hp/primera/restapi/mode/nodes.pm

```diff
@@ -0,0 +1,211 @@
+
+sub custom_node_output {
+    my ($self, %options) = @_;
+
+    return sprintf(
+            "node %s is %s",
+            $self->{result_values}->{id},
+            $self->{result_values}->{status}
+        );
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'global', type => 0, skipped_code => { -10 => 1 } },
+        { name => 'nodes', type => 1, message_multiple => 'All nodes are online.' },
+    ];
+
+    $self->{maps_counters}->{global} = [
+        {
+            label => 'total',
+            nlabel => 'nodes.total.count',
+            set => {
+                key_values => [ { name => 'total' } ],
+                output_template => 'Total number of nodes: %s',
+                perfdatas => [
+                    { template => '%s', min => 0 }
+                ]
+            }
+        },
+        {
+            label => 'online',
+            nlabel => 'nodes.online.count',
+            set => {
+                key_values => [ { name => 'online' }, { name => 'total' } ],
+                output_template => 'Number of online nodes: %s',
+                perfdatas => [
+                    { template => '%s', min => 0, max => 'total' }
+                ]
+            }
+        },
+        {
+            label => 'offline',
+            nlabel => 'nodes.offline.count',
+            set => {
+                key_values => [ { name => 'offline' }, { name => 'total' } ],
+                output_template => 'Number of offline nodes: %s',
+                perfdatas => [
+                    { template => '%s', min => 0, max => 'total' }
+                ]
+            }
+        }
+    ];
+
+    $self->{maps_counters}->{nodes} = [
+        {
+            label => 'node-status',
+            type => 2,
+            warning_default => '%{status} !~ /^online$/',
+            set => {
+                key_values => [ { name => 'status' }, { name => 'id' } ],
+                closure_custom_output => $self->can('custom_node_output'),
+                closure_custom_perfdata => sub { return 0; },
+                closure_custom_threshold_check => \&catalog_status_threshold_ng
+            }
+        }
+    ];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        'filter-node:s' => { name => 'filter_node' }
+    });
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+}
+
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $response = $options{custom}->request_api(
+        endpoint => '/api/v1/system'
+    );
+    my $total_nodes = 0;
+    my $online_nodes = 0;
+    my $offline_nodes = 0;
+
+    # Typical content of onlineNodes is [0, 1]
+    # %online_nodes_statuses associates 'online' to every online node is as key
```

Comment:
to every online node is as key -> to every online node is has key ?
I don't understand